### DISCUSSION TITLE
Is there a fastlane action for getting build details from testflight?

### DISCUSSION THREAD

**Discussion by SumanDas94** [fastlane/fastlane#20349] [Q&A] [answered]
Currently we can use `latest_testflight_build_number` to get the latest TestFlight build number
We have a use case where we would need the version numbers present in TestFlight or even better the latest actual version present in TestFlight
The closest thing i got is `fastlane pilot builds` but this can't be used directly in the fastfile I believe and it needs the apple account password to be given as input. 
Is there an action present or any in the roadmap which can get me the existing TestFlight versions or latest TestFlight version via the fastfile

**Comment by crazymanish** [comment, marked as answer]
Existing [latest_testflight_build_number](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/latest_testflight_build_number.rb) action provide both values [build version](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/latest_testflight_build_number.rb#L15) and [build number](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/latest_testflight_build_number.rb#L14). 

You should be able to fetch both values using the something like below code and we don't need another action or something to fetch the version.

```ruby
  lane :fetch_latest_testflight_build_version_and_number do
    # https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/latest_testflight_build_number.rb
    latest_testflight_build_number # you can provides more input params, if required.

    latest_build_version = lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]
    latest_build_number = lane_context[SharedValues::LATEST_TESTFLIGHT_BUILD_NUMBER]

    UI.important("The latest build number #{latest_build_number} of the latest version: #{latest_build_version}")
  end

**Reply by SumanDas94** [reply]
@crazymanish thank you for this. I was using `latest_testflight_build_number` with the version parameter and hence was getting the same value as latest version in the shared value which i passed as a param

But is the `lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]` set in context to the version number that is passed to the above action ?
Example
My latest testflight version is let's assume 175
`latest_testflight_build_number` -> sets `lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]` as 175 which is expected
` latest_testflight_build_number(version: '174')` -> sets `lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]` as 174 only

Is that what is expected given the shared value name refers to `LATEST_TESTFLIGHT_VERSION` and 174 isn't actually the latest version ?

**Reply by crazymanish** [reply]
- If you want the `latest version` and `latest build number` then please use `latest_testflight_build_number ` without version input param. 
- if you want the `latest build number` of a **specific version** then you have to pass the version input param.  i.e latest_testflight_build_number(version: '174') --> In this case, this action will return the latest build number of specific version 174 and sets lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION] as 174 also for further lanes, because you are doing something with specific 174 version, Fastlane will treat this as latest-version, which looks good/expected to me

To me, this action is working as expected so what is the question? 🤔

**Comment by SumanDas94** [comment]
> In this case, this action will return the latest build number of specific version 174 and sets lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION] as 174 also for further lanes, because you are doing something with specific 174 version

This has cleared my query. `lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]` this will be in context to the version i pass and if no version is passed, it will take latest. Got it thanks 👍

**Comment by SumanDas94** [comment]
@crazymanish sorry to open this discussion again but we are facing a scenario because of which we have a query 
We have following scenario
There are two threads on TF `1.0.180` , `1.0.181` . If the last uploaded build was on the thread `1.0.180` then is `latest_testflight_build_number` action supposed to return 1.0.180 as output for `lane_context[SharedValues::LATEST_TESTFLIGHT_VERSION]` even though `1.0.181` is the latest version thread on TF ?

**Reply by crazymanish** [reply]
if we will not use the explicit `version` inside the `latest_testflight_build_number` action then it will just return the details of the last/latest uploaded build.


logic is here for details: 
https://github.com/fastlane/fastlane/blob/9459056360450be4ac68962b70b6b9356046e65e/fastlane/lib/fastlane/actions/app_store_build_number.rb#L66-L88
